oauth2_http/java/com/google/auth/oauth2/DefaultCredentialsProvider.java

```diff
@@ -76,6 +76,9 @@
   static final String RUNTIME_JETTY_LOGGER = System.getProperty("org.eclipse.jetty.util.log.class");
   static final Logger LOGGER = Logger.getLogger(DefaultCredentialsProvider.class.getName());
 
+  /** Timeout in seconds to wait for the GCE metadata server when detecting the GCE environment. */
+  static final String GCE_METADATA_TIMEOUT_SECONDS_ENV_VAR = "GCE_METADATA_TIMEOUT";
```

Comment:
Let's confirm that this behavior and environment name will be consistent across all 7 auth libraries

Reply:
This is the env variable name and behavior as python (https://github.com/googleapis/google-auth-library-python/blob/master/google/auth/compute_engine/_metadata.py#L58). The rest auth libs are still using a default value. The behavior is already not consistent for more than 4 years.

This is a cheap fix to unblock GKE users while we figure out if we can do a better fix, there will be some work across 7 auth libs, either this fix or some better fix.